File: src/1_100/0083_Remove_Duplicates_from_Sorted_List/Remove_Duplicates_from_Sorted_List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if(head==null) {
            return head;
        }
        ListNode i = head;
        ListNode j = head;
        while(j != null) {
            if(i.val != j.val) {
                i = i.next;
                i.val = j.val;
            }
            j = j.next;
        }
        i.next = null;
        return head;
    }
}

File: src/1_100/0083_Remove_Duplicates_from_Sorted_List/Remove_Duplicates_from_Sorted_List.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
	def deleteDuplicates(self, head):
		"""
		:type head: ListNode
		:rtype: ListNode
		"""
		if not (head and head.nxt):
			return head
		i,j = head,head
		while j:
			if i.val!=j.val:
				i = i.next
				i.val = j.val
			j = j.next
		i.next = None
		return head
		
		
		
	# 不改变节点值	
	def deleteDuplicates(self, head):
		if not (head and head.next):
			return head
		i,j = head,head
		while i:
			if i.val!=j.val:
				j = j.next
			else:
				j.next = i.next
			i = i.next
		return head